src/DynamoRevit/DynamoRevitApp.cs

```diff
@@ -381,8 +381,41 @@ public static bool DynamoButtonEnabled
             set;
         }
 
-        private bool TryResolveDynamoCore()
+        /// <summary>
+        /// Whether the DynamoCore and DynamoRevit are in Revit's Addin, 
+        /// and Revit version is more than 2020
+        /// </summary>
+        /// <param name="application"></param>
+        /// <returns>True is that Dynamo and DynamoRevit are in Revit Addins</returns>
+        private static Boolean IsRevitInternalAddin(UIControlledApplication application)
+        {
+            if (application == null)
+                return false;
+            var revitVersion = application.ControlledApplication.VersionNumber;
+            if (int.Parse(revitVersion) < 2020)
+               return false;
+            var dynamoRevitRoot = Path.GetDirectoryName(Path.GetDirectoryName(assemblyName));
+            var RevitRoot = Path.GetDirectoryName(application.GetType().Assembly.Location);
+            if (dynamoRevitRoot.StartsWith(RevitRoot))
+            {
+                if (File.Exists(Path.Combine(dynamoRevitRoot, "DynamoInstallDetective.dll")) && File.Exists(Path.Combine(dynamoRevitRoot, "DynamoCore.dll")))
+                {
+                    var version_DynamoInstallDetective = FileVersionInfo.GetVersionInfo(Path.Combine(dynamoRevitRoot, "DynamoInstallDetective.dll"));
+                    var version_DynamoCore = FileVersionInfo.GetVersionInfo(Path.Combine(dynamoRevitRoot, "DynamoCore.dll"));
+                    if (version_DynamoCore.FileMajorPart == version_DynamoInstallDetective.FileMajorPart &&
+                        version_DynamoCore.FileMinorPart == version_DynamoInstallDetective.FileMinorPart &&
+                        version_DynamoCore.FileBuildPart == version_DynamoInstallDetective.FileBuildPart
+                       )
+                       return true;
+                }
+            }
+            return false;
+        }
+
+        private bool TryResolveDynamoCore(UIControlledApplication application)
         {
+            if(IsRevitInternalAddin(application))
+                dynamopath= Path.GetDirectoryName(Path.GetDirectoryName(assemblyName));
```

Comment:
just a style thing but I think this repo usually prefers braces when possible.

Reply:
I will add braces here.